.github/workflows/apps-cypress-percy-test.yml

```
@@ -10,45 +10,65 @@ on:
       - "apps/**"
     branches: [main]
 
+permissions:
+  contents: read
+
+concurrency:
+  group: site-e2e-${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
```

Concurrency group may cancel sibling matrix jobs. The group key does not include the app, so matrix jobs for the same ref share a group. With `cancel-in-progress: true` this can cause jobs to cancel each other on retries. Include the app:

```suggestion
concurrency:
  group: site-e2e-${{ github.workflow }}-${{ github.ref }}-${{ matrix.app }}
  cancel-in-progress: true
```